ref-farming/src/simple_farm.rs

```diff
@@ -292,17 +319,41 @@ impl SimpleFarm {
         (self.last_distribution.rps, claimed)
     }
 
-    pub fn can_be_removed(&self) -> bool {
+    /// Move an Ended farm to Cleared, if any unclaimed reward exists, go to beneficiary
+    pub(crate) fn move_to_clear(&mut self, total_seeds: &Balance) -> bool {
+        if let SimpleFarmStatus::Running = self.status {
+            self.distribute(total_seeds, true);
+        }
         if let SimpleFarmStatus::Ended = self.status {
-            if self.amount_of_claimed == self.amount_of_reward {
-                true
-            } else {
-                false
+            if self.last_distribution.unclaimed > 0 {
+                self.amount_of_claimed += self.last_distribution.unclaimed;
+                self.amount_of_beneficiary += self.last_distribution.unclaimed;
+                self.last_distribution.unclaimed = 0;
             }
+            self.status = SimpleFarmStatus::Cleared;
+            true
         } else {
             false
         }
     }
 
+    pub fn can_be_removed(&self, total_seeds: &Balance) -> bool {
+        match self.status {
+            SimpleFarmStatus::Ended => true,
```

Comment:
What about `::Cleared`?

Reply:
Now, `remove` means to move out of regular farms to outdated_farms, and ::Cleared is set during that moving. So, `can_be_removed` is targeted to those in regular farms, no ::Cleared one would exist.
To keep the outdated_farms in contract state is kind of a trade-off, to make it easy for frontend to get some history data.